src/plugins/data/server/data_views/fetcher/index_patterns_fetcher.ts

```diff
@@ -137,23 +137,21 @@ export class IndexPatternsFetcher {
   async validatePatternListActive(patternList: string[]) {
     const result = await Promise.all(
       patternList
-        .map((pattern) =>
-          this.elasticsearchClient.count({
-            index: pattern,
-          })
-        )
-        .map((p) =>
-          p.catch((e) => {
-            if (e.body.error.type === 'index_not_found_exception') {
-              return { body: { count: 0 } };
-            }
-            throw e;
-          })
-        )
+        .map(async (index) => {
+          const searchResponse = await this.elasticsearchClient.fieldCaps({
```

Comment:
ok, this is more complicated than it seems because the indices returned are actual index names vs the wildcard name. do you know a util function to match wildcard data?
```
patternList = [
  'apm-*-transaction*',
  'traces-apm*',
  'auditbeat-*',
  'endgame-*',
  'filebeat-*',
  'logs-*',
  'packetbeat-*',
  'winlogbeat-*',
  'ml_host_risk_score_*',
  '.siem-signals-default-*'
];
searchResponse.body.indices = [
  'apm-8.0.0-000001'-transaction-2021.09.30',
  'auditbeat-8.0.0-2021.09.30-000001', 
  '.siem-signals-default-000001' 
]
```

Reply:
I was looking at this code a little bit closer and I think I'm missing something:

`getFieldsForWildcard` calls `validatePatternListActive` to reduce the passed in array of patterns to only include ones that have indices. If that reduced list is empty or only a single pattern then it'll let `getFieldsForWildcard` determine if indices exist, if not that function will return an error (assuming `allow_no_indices` is false).

I think the piece I'm missing is why we need to call `validatePatternListActive`? If the caller of `getFieldsForWildcard` can pass in `allow_no_indices = true` to effectively do the same thing. I believe it should cause the function to not throw an error when one of the patterns results in no indices.

Here's an example:
![image](https://user-images.githubusercontent.com/56361221/135926295-3fadd52b-d35f-448f-95f2-8f0c79c06a5b.png)

Can we control `allow_no_indices` and remove `validatePatternListActive`?

Reply:
lets find out!

Reply:
ok, so i tried this out and its not going to work. we need to find out it NONE of the patterns are active because we do want to return an error then. so the check needs to happen before the FieldsForWildcard call

Reply:
> apm-*-transaction*

I think both of you are on something here, I think regex will be our friend here like `'apm-8.0.0-000001-transaction-2021.09.30'.match(/apm-.*-transaction.*/)` and we can find if the alias is matching one of the index returned.

what do you think?